xc/domain.ml

```diff
@@ -355,7 +355,13 @@ let make ~xc ~xs vm_info vcpus domain_config uuid final_uuid no_sharept =
   in
   debug "Domain_config: [%s]"
     (rpc_of arch_domainconfig domain_config |> Jsonrpc.to_string) ;
-  let domid = Xenctrl.domain_create xc config in
+  let domid =
+    if List.mem_assoc "domid" vm_info.platformdata then
```

Comment:
```
match List.assoc_opt "domid" vm_info.platformdata with
| None -> 0
| Some str -> int_of_string ste
```

Reply:
Another _option_ we're using quite a bit in xen-api:
```ocaml
List.assoc_opt "domid" vm_info.platformdata
|> Option.fold ~none:0 ~some:int_of_string
```

Reply:
`Option.fold ~none:0 ~some:int_of_string` looks intimidating but it conceptually simple. I wish it would be called `default`

Reply:
I guess `default` was not chosen because it could get confusing with the similar function `Option.value ~default:0 an_option`